src/com/google/enterprise/adaptor/DocIdSender.java

```diff
@@ -211,16 +216,64 @@ public DocId pushNamedResources(Map<DocId, Acl> resources,
     return null;
   }
 
+  @Override
   public GroupPrincipal pushGroupDefinitions(
       Map<GroupPrincipal, ? extends Collection<Principal>> defs,
-      boolean caseSensitive, ExceptionHandler handler) 
-      throws InterruptedException {
+      boolean caseSensitive, FeedType feedType, String groupSource,
+      ExceptionHandler handler) throws InterruptedException {
     if (config.markAllDocsAsPublic()) {
       log.finest("Ignoring attempt to send groups to the GSA because "
                  + "markAllDocsAsPublic is true.");
       return null;
     }
-    return pushGroupDefinitionsInternal(defs, caseSensitive, handler);
+    return pushGroupDefinitionsInternal(
+        defs, caseSensitive, feedType, groupSource, handler);
+  }
+
+  /*
+   * Issue: Full vs. Incremental Feeds for Groups.
+   *
+   * GSA 7.4.0 added support for full feeds of group definitions. A full
+   * feed replaces, rather than augments or updates, all groups from a
+   * given data source. This is useful because many connectors cannot
+   * detect when a group has been deleted, so repeated incremental updates
+   * would leave deleted groups in the GSA's groups database.
+   *
+   * When pushing a full feed, we must do it in a single "batch", because if
+   * done in batches and a later batch fails (or it simply takes a non-trivial
+   * amount of time to upload all the batches), we would end up with an
+   * incomplete set of groups.
+   *
+   * The problem with supplying all the group definitions in a single full
+   * feed is the GSA limit to the size of a feed file (1GB). Although this
```

Comment:
I saw such information in the documentation, but I didn't found any evidence of this in the code? Did you try to upload a big feed file e.g. 2-4 GB ?